src/voteHelper/VoteHelper.ts

```diff
@@ -0,0 +1,131 @@
+        const abstainVotes: Vote[] = [];
+        const referendaNumbers = referendums.map(referendum => referendum.number);
+
+        for (const account of network.votersAccounts) {
+            const voterVotes = await this.getAccountVotes(account);
+            const relevantVotes = voterVotes.filter(vote => referendaNumbers.includes(vote.referendaNumber));
+
+            for (const vote of relevantVotes) {
+                if (!uniqueVotes.has(vote.referendaNumber)) {
+                    uniqueVotes.add(vote.referendaNumber);
+                    newVoteForReferenda.push(vote);
+                }
+            }
+        }
+        
+        for (const referenda of referendums) {
+            if (!uniqueVotes.has(referenda.number)) {
+                // Create an abstain vote for the referenda
+                const abstainVote = new AbstainVote({
+                    referendaNumber: referenda.number, data: {
+                        SplitAbstain: {
+                            aye: "0",
+                            nay: "0",
+                            abstain: network.voteAmount,
+                        },
+                    }
+                });
+                abstainVotes.push(abstainVote);
+            }
+        }
+
+        // Combine the new votes and abstain votes
+        const combinedVotes = [...newVoteForReferenda, ...abstainVotes];
+
+        return combinedVotes;
+    }
+
+    async decideHowToVote(validator: string, potentialVotes: Vote[]): Promise<Vote[]> {
+        const validatorVotes = await this.getAccountVotes(validator);
+        return potentialVotes.filter(potentialVote => {
+            const validatorVote = validatorVotes.find(v => v.referendaNumber == potentialVote.referendaNumber);
+            return this.compareVotes(validatorVote, potentialVote);
+        });
+    }
+
+    compareVotes(validatorVote: Vote | undefined, potentialVote: Vote): boolean {
+        if (validatorVote instanceof AbstainVote) {
+            if (potentialVote instanceof AbstainVote) {
+                // if validatorVote and potentialVote is abstain - do nothing
+                return false;
+            } else {
+                // if validatorVote is abstain and potentialVote is NOT abstain - change vote
+                return true;
+            }
+        } else if (validatorVote instanceof StandardVote) {
+            if (potentialVote instanceof AbstainVote) {
+                // if validatorVote is standard and potentialVote is abstain - change vote
+                return true;
+            } else if (potentialVote instanceof StandardVote) {
+                // if validatorVote is standard and potentialVote is standard, then check direction if direction is different - change the vote
+                return validatorVote.direction != potentialVote.direction;
+            } else {
+                return validatorVote.direction != potentialVote.direction;
+            }
+        }
+        return true;
+    }
```

Comment:
I think this could be simplified to something like:

``` typescript
if (typeof validatorVote != typeOf potentialVote) return true;
if (validatorVote is AbstainVote and potentialVote is AbstainVote) return false;
if (validatorVote is StandardVote and potentialVote is StandardVote) return validatorVote.direction != potentialVote.direction;
return true;
```

Reply:
thank you for that, looks really cool!